Add perspective-project-bridge

arunkmv opened the pull request:
### Brief summary of what the package does
This package adds the missing bridge between perspective.el and project.el.

It creates a perspective for each project.el project. Based on [persp-mode-projectile-bridge](https://github.com/Bad-ptr/persp-mode-projectile-bridge.el).

### Direct link to the package repository

https://github.com/arunkmv/perspective-project-bridge

### Your association with the package

I am the maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [ ] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

**NOTE:** I've built the package successfully, but installing it gives the error ```Package does not untar cleanly into directory perspective-project-bridge-20231021.1853/```. Manually untarring it works fine. Please let me know what the issue could be.

riscy commented:
Thank you for this.  I suppose this is similar to [persp-projectile](https://melpa.org/#/persp-projectile) and maybe you should take a look at that package to see if there are any alignments you could make.  In the meantime, a quick first pass --

`perspective-project-bridge.el` with byte-compile using Emacs 28.1:
```
perspective-project-bridge.el:80:18: Warning: reference to free variable
    `perspective-project-bridge-mode'
In perspective-project-bridge-find-perspectives-for-all-buffers:
perspective-project-bridge.el:92:9: Warning: reference to free variable
    `perspective-project-bridge-mode'
In end of data:
perspective-project-bridge.el:83:23: Warning: the function `project-name' is
    not known to be defined.
```

`perspective-project-bridge.el` with package-lint 20230905.629:
```
3 issues found:
6:73: warning: An explicit dependency on cl-lib <= 1.0 is not needed on Emacs >= 24.3.
8:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
83:15: error: You should depend on (emacs "29.1") if you need `project-name'.
```

`perspective-project-bridge.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
```
(Note that downstreams tools won't know how to "rewrap" this text, so it's best to do it on your end)

arunkmv commented:
Hi!

Thank you for all the feedback.

Yes, I am aware of persp-projectile, in fact I was using it till last week!

Perspective and persp-mode are two very popular workspace utilities, while projectile and the native project.el are the de-facto project managers. 3 "bridges" for these package exist:

- [persp-projectile](https://melpa.org/#/persp-projectile)
- [persp-mode-project-bridge](https://melpa.org/#/persp-mode-project-bridge)
- [persp-mode-projectile-bridge](https://melpa.org/#/persp-mode-projectile-bridge)

I switched to project.el recently, but wanted functionality similar to persp-projectile. I quickly found out that no such package exists (atleast to my knowledge), so I attempted to write my own. I liked the approach the persp-mode based bridge packages took using a minor-mode and adding hooks so I decided to try that. This package now adds the fourth missing link, between perspective and project.el.

As for your feedback on the code, I have added the necessary fixes and have checked with package-lint. I was not aware of this package, thank you for introducing me to it.

Please let me know if you find any further issues. Thanks again!

riscy commented:
Thank you!  Just one note to be aware of - if you ever move this package to depend on Emacs 28, I believe the following will apply:

`perspective-project-bridge.el` with byte-compile using Emacs 28.1:
```
perspective-project-bridge.el:92:32: Warning: `project-roots' is an obsolete
    function (as of 0.3.0); use `project-root' instead.
```

...and, welcome to MELPA. :)

arunkmv commented:
Awesome! Thank you for the heads up.
